src/main/java/com/softserve/app/service/UserService/UserServiceImpl.java

```diff
@@ -57,68 +57,55 @@ public User findByEmail(String email) {
 
-        // Username
-        if (userDTO.getUsername() != null) {
-            curUser.setUsername(userDTO.getUsername());
-        }
-
-        // Email
-        if (userDTO.getId().equals(curUser.getId())) {
-            curUser.setEmail(userDTO.getEmail());
-        }
-
-        // Photo = avatar
-        if (userDTO.getPhotoUrl() != null) {
-            curUser.setPhotoUrl(userDTO.getPhotoUrl());
-        }
+        // Avatar
+        if (file != null) {
+            dto.setPhotoUrl(fileService.saveImg(file));
+        } else dto.setPhotoUrl(usrFromDb.getPhotoUrl());
 
         // Password
-        if (userDTO.getPassword() != null) {
-            curUser.setPassword(userDTO.getPassword());
-        }
-/*
-        // Surveys list
-        if (userDTO.getUserSurveys() != null) {
-            curUser.setUserSurveys(new HashSet<>(userDTO.getUserSurveys()));
-        }
+        if ((dto.getPassword() != null) && (dto.getNew_pass() != null)) {
+            String encodedNew = checkPassword(usrFromDb, dto.getPassword(), dto.getNew_pass(), dto.getNew_pass_2());
+            if (encodedNew != null) {
+                dto.setPassword(encodedNew);
+            }
+        } else dto.setPassword(usrFromDb.getPassword());
+
+        userRepository.save(User.builder()
+                .id(dto.getId())
+                .username(dto.getUsername() != null ? dto.getUsername() : usrFromDb.getUsername())
+                .email(dto.getEmail() != null ? dto.getEmail() : usrFromDb.getEmail())
+                .photoUrl(dto.getPhotoUrl())
+                .password(dto.getPassword())
+                .build());
 
-        // Favourites list
-        if (userDTO.getFavourites() != null) {
-            curUser.setFavourites(new HashSet<>(userDTO.getFavourites()));
-        }
-*/
-        return curUser.ofDTO();
     }
 
 
-    @Value("${img.path}")
-    private String uploadPath;
-
     @Override
-    public UserDTO updateAvatar(Long user_id, MultipartFile userAva) {
-        User user = userRepository.findById(user_id).orElseThrow(() -> new SportHubException(
+    public void deleteUser(User user) {
+        User usr = userRepository.findById(user.getId()).orElseThrow(() -> new SportHubException(
                 SportHubConstant.USER_NOT_FOUND.getMessage(), 404));
+        System.out.println("User service: deleting user " + usr.getId());
 
+        userRepository.delete(usr);
+    }
 
-        String directory = uploadPath + "UserAvatars/" + user.getId() + "avatar.jpg";
 
-        File logo = new File(directory);
-        try {
-            userAva.transferTo(logo);
-            user.setPhotoUrl(logo.getName());
-        } catch (Exception e) {
-            throw new SportHubException(SportHubConstant.FILE_LOADING_EXCEPTION.getMessage(), 500);
+    private String checkPassword(User user, String oldPass, String pass1, String pass2) {
+        if (pass1.equals(pass2)) {
+            BCryptPasswordEncoder passwordEncoder = new BCryptPasswordEncoder();
```

Comment:
you have a bean of password encoder : https://github.com/nazar-goncharenko/SoftServe-course-backend/blob/f924162fabfdcca4a8750403e8911e8403702238/src/main/java/com/softserve/app/config/SpringWebSecurityConfig.java#L38, just inject it and use.